src/main/scala/srdc/smartcds/cds/flow/ACCAHAFlowExecution.scala

```diff
@@ -0,0 +1,395 @@
+                  HypertensiveTreatment: Seq[MedicationStatement], Ethnicity: Seq[Observation],
+                  responseBuilder: CdsResponseBuilder): CdsResponseBuilder = {
+
+    var output = responseBuilder
+    val riskScores = calculateACCRisk(patient, TotalCholesterol, HDLCholesterol, SystolicBP, SmokingStatus,
+      Type1Diabetes, Type2Diabetes, HypertensiveTreatment, Ethnicity)
+
+    riskScores match {
+      case Some((patientScore, healthyScore)) =>
+        output = output.withCard(_.loadCardWithPostTranslation("card-score",
+          "effectiveDate" -> DateTimeUtil.zonedNow(),
+          "patientScoreValue" -> patientScore,
+          "healthyScoreValue" -> healthyScore
+        ))
+        output = recommendStopSmokingIfApplicable(SmokingStatus, output)
+        output = recommendReduceBPIfApplicable(SystolicBP, output)
+    }
+
+    output
+  }
+
+  /**
+   * Recommends blood pressure reduction in case of need
+   *
+   * @param BP_SBP Systolic Blood Pressure of the patient
+   * @param output CdsResponseBuilder object
+   * @return whether or not SBP is too high and should be reduced
+   */
+  private def recommendReduceBPIfApplicable(BP_SBP: Seq[Observation], output: CdsResponseBuilder) = {
+    val sbp = FhirParseHelper.getSystolicBP(BP_SBP).get
+    if (sbp > 140) {
+      output.withCard(_.loadCardWithPostTranslation("card-reduce-bp",
+        "effectiveDate" -> DateTimeUtil.zonedNow()
+      ))
+    } else {
+      output
+    }
+  }
+
+  /**
+   * Recommends whether patient should stop smoking or not
+   *
+   * @param SmokingStatus patient's current smoking status
+   * @param output        CdsResponseBuilder object
+   * @return whether or not patient should stop smoking
+   */
+  private def recommendStopSmokingIfApplicable(SmokingStatus: Seq[Observation], output: CdsResponseBuilder): CdsResponseBuilder = {
+    /*
+    I modified the condition because my algorithm only cares whether you
+    smoke or not, and my determineSmokingStatus function returns 1 if patient
+    is considered a smoker and 0 otherwise.
+    */
+    if (determineSmokingStatus(SmokingStatus.headOption) > 0) {
+      output.withCard(_.loadCardWithPostTranslation("card-stop-smoking",
+        "effectiveDate" -> DateTimeUtil.zonedNow()
+      ))
+    } else {
+      output
+    }
+  }
+
+  /**
+   * Checks whether the needed resource is present or not
+   *
+   * @param resources Resource to check the state of sequences
+   * @return 1 if nonempty 0 otherwise
+   */
+  def checkExists(resources: Seq[Any]): Int = if (resources.nonEmpty) 1 else 0
+
+  /**
+   * Determines whether patient has Hypertensive Treatment or not
+   *
+   * @param medications sequence of patient's Medication Statements
+   * @return 1 if patient is under treatment, 0 otherwise
+   */
+  private def checkHypertensiveTreatment(medications: Seq[MedicationStatement]): Int = {
+    val hypertensiveCodes = Set("C02", "C03", "C07", "C08", "C09")
```

Comment:
Do not define codes manually, get the codes from the bundles instead.

Reply:
Removed checkHypertensiveTreatment.